.semaphore/semaphore.yml

```diff
@@ -6,33 +6,50 @@ agent:
     os_image: ubuntu1804
 
 global_job_config:
+  env_vars:
+    - name: CACHE_BASE_DIR
+      value: cache
   prologue:
     commands:
       - checkout --use-cache
       - export DOCKER_IMAGE_TAG=${SEMAPHORE_GIT_TAG_NAME:-${SEMAPHORE_GIT_SHA}}
+      - echo CACHE_BASE_DIR=${CACHE_BASE_DIR}
       - echo DOCKER_IMAGE_TAG=${DOCKER_IMAGE_TAG}
       - echo SEMAPHORE_GIT_SHA=${SEMAPHORE_GIT_SHA}
+      - echo SEMAPHORE_GIT_TAG_NAME=${SEMAPHORE_GIT_TAG_NAME}
 
 blocks:
   - name: Build
+    dependencies: []
     task:
       jobs:
         - name: Build Docker Image
           commands:
-            - make docker/build
-            - make semaphore/store
+            - if ! cache has_key ${SEMAPHORE_GIT_SHA} ; then
+                make docker/build ;
+                make docker/save ;
+                cache store ${SEMAPHORE_GIT_SHA} ${CACHE_BASE_DIR} ;
+              fi
+        - name: Check Versions
+          commands:
+            - make check/updates
 
   - name: Forensics
+    dependencies: [ "Build" ]
     task:
       secrets:
         - name: SNYK_CREDENTIALS
       prologue:
         commands:
-          - make semaphore/restore
+          - cache restore ${SEMAPHORE_GIT_SHA}
+          - make docker/load
       jobs:
         - name: Run Snyk
           commands:
             - make test/snyk
+        - name: Run Tools
+          commands:
+            - make test/execute-tools
 
 promotions:
```

Comment:
hm, somehow, the promotion runs twice now

<img width="1111" alt="Screenshot 2020-05-17 at 09 59 27" src="https://user-images.githubusercontent.com/930643/82140211-fb90bc80-9825-11ea-9139-280bfc784971.png">

Reply:
started it twice to see what happens...

Reply:
ah gotcha!